src/main/java/uk/gov/hmcts/darts/datamanagement/service/impl/InboundToUnstructuredProcessorSingleElementImpl.java

```diff
@@ -226,6 +227,9 @@ private void performValidation(
         if (fileSize > maxFileSize) {
             unstructured.setStatus(getStatus(FAILURE_FILE_SIZE_CHECK_FAILED));
         }
+        if (0 == fileSize) {
+            unstructured.setStatus(getStatus(FAILURE_EMPTY_FILE));
+        }
```

Comment:
This is more of a question on the existing structure of the `performValidation()` method. The way the conditionals are structured essentially makes each condition fall-through. So if a file is of the wrong format, and has an unexpected size then `unstructured.status` is being overwritten, and all we're only left with the last status value set.

I find that a bit counterintuitive - is that an intentional design choice, or should the method return on the first validation failure?

Or if we're happy to only set the last failure reason, should we at least log any proceeding failure reasons?